File: archive/c/c/capitalize.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

char *captialize(char str[]) {
    for(int i = 0; i < strlen(str); i++) {
        if(i == 0) {
            str[i] = toupper(str[i]);
        } else {
            continue;
        }
    }
    return str;
}

int main(int argc, char *argv[]) {
    if(argc == 2 && strlen(argv[1]) != 0) {
        printf("%s\n", captialize(argv[1]));
    } else if(argc > 2) {
        printf("Use quotes around multiple strings.\n");
    } else {
        printf("Usage: please provide a string\n");
    }

    return 0;
}

File: archive/g/go/capitalize.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func exitWithError() {
	fmt.Println("Usage: provide a string")
	os.Exit(1)
}

func uppercaseFirst(str string) string {
	s := string(str[0])
	u := strings.ToUpper(s)
	up := u + str[1:]
	return up
}

func main() {
	if len(os.Args) != 2 {
		exitWithError()
	}

	s := os.Args[1]
	up := uppercaseFirst(s)

	fmt.Println(up)
}
